### ISSUE TITLE
type.late

### ISSUE THREAD

**Issue by matthewp** [canjs/can-type#19]
# type.late

This new API is used to create a TypeObject that will be defined at a later time. It needs to convey the same API as a TypeObject (`can.isMember` and `can.new` symbols) but also a way to "unwrap" the TypeObject and returns another underlying TypeObject.

With this change I propose the following new APIs.

## type.late(fn)

A function that takes a function and returns a TypeObject.

```js
let typeObject = type.late(() => type.maybe(String));
```

Questions:

1. What guarantees about timing should type.late have?

## type.isTypeObject(Type)

Currently there is no great way to know if an object is a TypeObject. In places like can-observable-mixin it will manually check for the required symbols. It makes sense to have a function in can-type that does this work.

```js
class Faves extends DefineObject {}

type.isTypeObject(Faves); // -> true

type.isTypeObject({
  [Symbol.for("can.isMember")]() {}
  [Symbol.for("can.new")]() {}
}); // -> true

type.isTypeObject(Date); // false
```

## type.normalize(Type)

Takes any `Type` and returns a TypeObject. Type could be:
  * A builtin constructor like `Date` or `Number`.
  * A TypeObject that implements `can.unwrapType`.
  * A TypeObject.

This function takes all of these things and returns A TypeObject that can be used with `canReflect.convert`.

Questions:

1. Should `normalize()` unwrap late TypeObjects? Otherwise there could be another method that does this. The primary use-case I have in mind for `type.normalize` is as part of can-observable-mixin which builds up a property definition and currently does the equivalent of type.normalize itself. In this case it would *want* to have a unwrapped version.

**Comment by matthewp** [comment]
These should be equivalent:

* `type.late(() => String)`
* `type.late(() => type.check(String))`

**Comment by matthewp** [comment]
Implemented by #21
